domain/src/test/java/org/oppia/android/domain/topic/TopicListControllerTest.kt

```diff
@@ -351,22 +379,167 @@ class TopicListControllerTest {
     )
     testCoroutineDispatchers.runCurrent()
 
-    storyProgressController.recordRecentlyPlayedChapter(
+    storyProgressController.recordCompletedChapter(
       profileId0,
       RATIOS_TOPIC_ID,
-      RATIOS_STORY_ID_1,
-      RATIOS_EXPLORATION_ID_2,
+      RATIOS_STORY_ID_0,
+      RATIOS_EXPLORATION_ID_1,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    val promotedActivityList = retrievePromotedActivityList()
+    assertThat(promotedActivityList.promotedStoryList.suggestedStoryCount)
+      .isEqualTo(1)
+    verifyPromotedStoryAsFractionStory0Exploration0(
+      promotedActivityList.promotedStoryList.suggestedStoryList[0]
+    )
+  }
+
+  @Test
+  fun testGetStoryList_markStoryDoneOfRatiosAndFirstTestTopic_suggestedStoryListIsCorrect() {
+    storyProgressController.recordCompletedChapter(
+      profileId0,
+      TEST_TOPIC_ID_0,
+      TEST_STORY_ID_0,
+      TEST_EXPLORATION_ID_2,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    storyProgressController.recordCompletedChapter(
+      profileId0,
+      TEST_TOPIC_ID_0,
+      TEST_STORY_ID_0,
+      TEST_EXPLORATION_ID_5,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    storyProgressController.recordCompletedChapter(
+      profileId0,
+      RATIOS_TOPIC_ID,
+      RATIOS_STORY_ID_0,
+      RATIOS_EXPLORATION_ID_0,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    storyProgressController.recordCompletedChapter(
+      profileId0,
+      RATIOS_TOPIC_ID,
+      RATIOS_STORY_ID_0,
+      RATIOS_EXPLORATION_ID_1,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    val promotedActivityList = retrievePromotedActivityList()
+
+    assertThat(promotedActivityList.promotedStoryList.suggestedStoryCount)
+      .isEqualTo(2)
+    verifyPromotedStoryAsSecondTestTopicStory0Exploration0(
+      promotedActivityList.promotedStoryList.suggestedStoryList[0]
+    )
+    verifyPromotedStoryAsFractionStory0Exploration0(
+      promotedActivityList.promotedStoryList.suggestedStoryList[1]
+    )
```

Comment:
I think this is wrong. If test topic 0 is finished, then fractions shouldn't be recommended since it's a prerequisite for that topic. Only the second test topic should be promoted in this case.

Reply:
> I think this is wrong. If test topic 0 is finished, then fractions shouldn't be recommended since it's a prerequisite for that topic. Only the second test topic should be promoted in this case.

@Ben can you please clarify here 
Finishing " First Test Topic" recommend "Second Test Topic"
As per this comment https://github.com/oppia/oppia-android/pull/2272#discussion_r567145591
Finishing " First Test Topic" recommend "Ratios"

Reply:
@veena14cs updated my comment to be less ambiguous. Here's another view of it:
- If the user finishes no topics, suggest fractions + ratios
- If the user completes only ratios, suggest fractions
- If the user completes only fractions, suggest first test topic + ratios
- If the user completes the first test topic, suggest ratios
- If the user completes fractions + ratios, suggest the first test topic
- If the user completes fractions + the first test topic + ratios, suggest second test topic
- If the user completes first test topic + ratios, suggest second test topic
- If the user completes any prerequisites + the second test topic, show upcoming

I believe that covers all possible cases in the test graph (though the last point is simplifying a bunch of different variations on that one).

Reply:
3nd point  - `If the user completes only fractions, suggest first test topic + ratios `
It has to be both First test topic and Ratios or only recommend First test topic?

Reply:
Both first topics & ratios since, like the first case, there is no prerequisite for ratios and ratios hasn't been completed, so we recommend it.

The second point is correct for the same reason: there is no prerequisite for fractions so we always recommend it unless it's completed or a later topic that depends on it is completed (e.g. first test topic).